File: backend/app/schemas/ideas.py
from __future__ import annotations
from typing import Any, Literal, Optional
from pydantic import BaseModel, Field

IdeaCategory = Literal[
    "process_improvement", "technology", "cost_saving",
    "talent_development", "client_delivery", "workplace_culture", "other"
]
EvaluationStatus = Literal["submitted", "under_review", "accepted", "rejected"]


class AttachmentInfo(BaseModel):
    id: str
    name: str
    size: int
    mime_type: str
    is_image: bool


class EvaluationInfo(BaseModel):
    status: EvaluationStatus
    comment: Optional[str] = None
    evaluated_at: Optional[str] = None
    assigned_admin_id: Optional[str] = None
    assigned_admin_name: Optional[str] = None


class EvaluateIdeaRequest(BaseModel):
    status: EvaluationStatus
    comment: Optional[str] = Field(None, max_length=1000)


class IdeaDetailResponse(BaseModel):
    id: str
    title: str
    description: str
    category: str
    submitter_id: str
    submitter_name: str
    submitted_at: str
    attachments: list[AttachmentInfo] = []
    evaluation: EvaluationInfo
    extra_data: Optional[dict[str, Any]] = None


class IdeaSummaryResponse(BaseModel):
    id: str
    title: str
    category: str
    submitter_name: str
    submitted_at: str
    attachment_count: int
    evaluation_status: EvaluationStatus
    reviewer_name: Optional[str] = None
    extra_data: Optional[dict[str, Any]] = None


class IdeaListResponse(BaseModel):
    ideas: list[IdeaSummaryResponse]
    total: int
    page: int
    limit: int


class IdeaCreateRequest(BaseModel):
    title: str = Field(..., min_length=1, max_length=150)
    description: str = Field(..., min_length=1, max_length=3000)
    category: IdeaCategory

File: frontend/src/types/ideas.ts
export interface AttachmentInfo {
  id: string
  name: string
  size: number
  mime_type: string
  is_image: boolean
}

export type EvaluationStatus = 'submitted' | 'under_review' | 'accepted' | 'rejected'

export interface EvaluationInfo {
  status: EvaluationStatus
  comment: string | null
  evaluated_at: string | null
  assigned_admin_id: string | null
  assigned_admin_name: string | null
}

export interface EvaluateIdeaRequest {
  status: EvaluationStatus
  comment?: string
}

export interface IdeaDetailResponse {
  id: string
  title: string
  description: string
  category: string
  submitter_id: string
  submitter_name: string
  submitted_at: string
  attachments: AttachmentInfo[]
  evaluation: EvaluationInfo
  extra_data: Record<string, unknown> | null
}

export interface IdeaSummaryResponse {
  id: string
  title: string
  category: string
  submitter_name: string
  submitted_at: string
  attachment_count: number
  evaluation_status: EvaluationStatus
  reviewer_name: string | null
  extra_data: Record<string, unknown> | null
}

export interface IdeaListResponse {
  ideas: IdeaSummaryResponse[]
  total: number
  page: number
  limit: number
}
